diff --git a/shiehnews/protected/controllers/AdminController.php b/shiehnews/protected/controllers/AdminController.php
index a31c174..e68d2b2 100644
--- a/shiehnews/protected/controllers/AdminController.php
+++ b/shiehnews/protected/controllers/AdminController.php
@@ -126,15 +126,14 @@ class AdminController extends BaseController {
 					break;
 				case 'update':
 					if (isset($_GET['ID'])) {
-						$form = new Category;
+						$category = Category::model()->findByPk($_GET['ID']);
 						if (isset($_POST['Category'])) {
-							$form->attributes = $_POST['Category'];
-							if ($form->save()) {
+							$category->attributes = $_POST['Category'];
+							if ($category->save()) {
 								$this->redirect(array('admin/categories', 'action' => 'list'));
 							}
 						}
-						
-						$this->render('category/update', array('form' => $form));
+						$this->render('category/edit', array('form' => $category));
 					} else
 						throw new CHttpException(404, "您所访问的地址并不存在!");
 					break;
diff --git a/shiehnews/protected/models/Category.php b/shiehnews/protected/models/Category.php
index e6c1689..cb8b489 100644
--- a/shiehnews/protected/models/Category.php
+++ b/shiehnews/protected/models/Category.php
@@ -25,6 +25,7 @@ class Category extends CActiveRecord
 	public function rules()
 	{
 		return array(
+			array('name', 'required'),
 		);
 	}
 
@@ -58,4 +59,4 @@ class Category extends CActiveRecord
 	public function getUrl() {
 		return 'category/' . $this->id . '.html';
 	}
-}
\ No newline at end of file
+}
diff --git a/shiehnews/protected/views/admin/category/editCategory.php b/shiehnews/protected/views/admin/category/editCategory.php
deleted file mode 100644
index 398bdb2..0000000
--- a/shiehnews/protected/views/admin/category/editCategory.php
+++ /dev/null
@@ -1,3 +0,0 @@
-<?php $this->renderPartial('_categoryForm', array(
-	'category' => $category,
-)); ?>
\ No newline at end of file
diff --git a/shiehnews/protected/views/admin/category/list.php b/shiehnews/protected/views/admin/category/list.php
index 63f6d2f..c6a3d2b 100644
--- a/shiehnews/protected/views/admin/category/list.php
+++ b/shiehnews/protected/views/admin/category/list.php
@@ -9,8 +9,8 @@
 			<td>#<?php echo $key + 1; ?></td>
 			<td><?php echo $model->name; ?></td>
 			<td>
-				<?php echo CHtml::link('修改', array('admin/categories', 'action' => 'update', 'cID' => $model->id)); ?> | 
-				<?php echo CHtml::link('删除', array('admin/deleteCategory', 'action' => 'delete', 'cID' => $model->id)); ?>
+				<?php echo CHtml::link('修改', array('admin/categories', 'action' => 'update', 'ID' => $model->id)); ?> | 
+				<?php echo CHtml::link('删除', array('admin/deleteCategory', 'action' => 'delete', 'ID' => $model->id)); ?>
 			</td>
 		</tr>
 	<?php endforeach; ?>
